vowpalwabbit/cache.cc

```diff
@@ -70,98 +70,102 @@ void VW::write_example_to_cache(io_buf& output, example* ae, label_parser& lbl_p
     VW::details::cache_temp_buffer& temp_buffer)
 {
   temp_buffer._backing_buffer->clear();
-  lbl_parser.cache_label(ae->l, ae->_reduction_features, temp_buffer._temporary_cache_buffer, "", false);
-  cache_features(temp_buffer._temporary_cache_buffer, ae, parse_mask);
-  temp_buffer._temporary_cache_buffer.flush();
+  io_buf& temp_cache = temp_buffer._temporary_cache_buffer;
+  lbl_parser.cache_label(ae->l, ae->_reduction_features, temp_cache, "_label", false);
+  cache_tag(temp_cache, ae->tag);
+  temp_cache.write_value<unsigned char>(ae->is_newline ? newline_example : non_newline_example);
+  temp_cache.write_value<unsigned char>(static_cast<unsigned char>(ae->indices.size()));
```

Comment:
Maybe add an assert here that it somehow this value is not larger

Reply:
Larger than what?

Reply:
Than an unsigned char

Reply:
Added